src/mgmt/srv_query.c

```diff
@@ -224,32 +224,73 @@ ds_mgmt_bio_health_query(struct mgmt_bio_health *mbh, uuid_t dev_uuid,
 	return rc;
 }
 
+struct bio_list_devs_info {
+	d_list_t	dev_list;
+	int		dev_list_cnt;
+};
+
+static void
+bio_query_dev_list(void *arg)
+{
+	struct bio_list_devs_info	*list_devs_info = arg;
+	struct dss_module_info		*info = dss_get_module_info();
+	struct bio_xs_context		*bxc;
+	int				 rc;
+
+	D_ASSERT(info != NULL);
+
+	bxc = info->dmi_nvme_ctxt;
+	if (bxc == NULL) {
+		D_ERROR("BIO NVMe context not initialized for xs:%d, tgt:%d\n",
+			info->dmi_xs_id, info->dmi_tgt_id);
+		return;
+	}
+
+	rc = bio_get_dev_list(&list_devs_info->dev_list,
+			      &list_devs_info->dev_list_cnt, bxc);
```

Comment:
Call bio_dev_list() directly here if you use dss_ult_execute().

Reply:
done